src/pages/Main/MainProduct/MainProduct.js

```diff
@@ -1,48 +1,72 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
 
 const MainProduct = () => {
-  const [mainProductData, setMainProductData] = useState([]);
+  const [mainProductList, setMainProductList] = useState([]);
   const navigate = useNavigate();
+  const obsTarget = useRef(null);
 
   const goToShop = () => {
     navigate('/shop');
   };
 
   useEffect(() => {
-    fetch('/data/productData.json', {
-      method: 'GET',
-    })
+    fetch('/data/productData.json')
       .then(res => res.json())
       .then(data => {
-        setMainProductData(data);
+        setMainProductList(data);
       });
   }, []);
 
+  useEffect(() => {
+    const io = new IntersectionObserver(entry => {
+      const target = entry[0];
+      if (target.isIntersecting) {
+        fetch('/data/productData.json')
+          .then(res => res.json())
+          .then(result => {
+            setMainProductList(prev => [...prev, ...result]);
+          });
+      }
+    });
```

Comment:
어차피 사용되는건 isIntersecting 밖에 없으니 callback에서 보내줄 때 바로 분해할당 시키는건 어때요?
```suggestion
    const io = new IntersectionObserver([{ isIntersecting }] => {
      if (isIntersecting) {
        fetch('/data/productData.json')
          .then(res => res.json())
          .then(result => {
            setMainProductList(prev => [...prev, ...result]);
          });
      }
    });
```

Reply:
넵 감사합니다 ! :)